src/dune_rules/format_rules.ml

```diff
@@ -133,16 +147,17 @@ let gen_rules_output
         Source_tree.Dir.dune_file source_dir
         |> Memo.Option.iter ~f:(fun f ->
           Dune_file0.path f
-          |> Memo.Option.iter ~f:(fun path ->
-            let input_basename = Path.Source.basename path in
-            let input = Path.build (Path.Build.relative dir input_basename) in
-            let output = Path.Build.relative output_dir input_basename in
-            (let open Action_builder.O in
-             let+ () = Action_builder.path input in
-             Action.Full.make (action ~version input output))
-            |> Action_builder.with_file_targets ~file_targets:[ output ]
-            |> Super_context.add_rule sctx ~mode:Standard ~loc ~dir
-            >>> add_diff sctx loc alias_formatted ~dir ~input ~output)))
+          |> Memo.Option.iter
+               ~f:(add_diff_rule ~sctx ~loc ~alias_formatted ~dir ~output_dir ~version)))
+  and* () =
+    match Format_config.includes config Dune with
+    | false -> Memo.return ()
+    | true ->
+      Dune_project.file project
+      |> Memo.Option.iter ~f:(fun path ->
+        if String.equal "_build/default" (Path.Build.to_string dir)
```

Comment:
I'm not happy with this if check for the hard-coded build directory, but I didn't find a way to check if we are in the "_build/default" directory other than this.

Reply:
How about checking it for equality against `Context.build_dir`?